src/main/java/objective/taskboard/controller/AvatarController.java

```
@@ -65,22 +42,31 @@
-            Resource resource = IOUtilities.asResource(url);
             if (url.toString().contains(jiraProperties.getUrl()))
-                resource = IOUtilities.asResource(jiraEndpointAsUser.readBytesFromURL(url));
-
-            response.setHeader("Cache-Control", "private, max-age=21600");
-            response.setHeader("Pragma", "cache");
-            return ResponseEntity.ok()
-                    .contentLength(resource.contentLength())
-                    .header("Content-Type", contentType)
-                    .body(resource);
-        } catch (Exception e) {
-            log.warn("Error getting avatar", e);
-            return ResponseEntity.status(INTERNAL_SERVER_ERROR)
-                    .header("Content-Type", "text/html; charset=utf-8")
-                    .body(defaultIfEmpty(e.getMessage(), e.toString()));
+            	return responseFromUrl(IOUtilities.asResource(jiraEndpointAsUser.readBytesFromURL(url)), contentType, response);
+            
+            return responseFromUrl(IOUtilities.asResource(IOUtils.toByteArray(url.openStream())), contentType, response);
+        } catch (Exception e) {//NOSONAR
+            log.warn("Error getting avatar for " + username + " " + e.getMessage());
+            return responseFromUrl(
+            		IOUtilities.asResource(AvatarController.class.getResource("/static/images/userwithoutavatar.svg")),
+            		"image/svg+xml",
+            		response);
         }
     }
 
+    private ResponseEntity<Object>  responseFromUrl(Resource resource, String contentType, HttpServletResponse response) {
+    	try {
+	        response.setHeader("Cache-Control", "private, max-age=21600");
+	        response.setHeader("Pragma", "cache");
+	        return ResponseEntity.ok()
+	                .contentLength(resource.contentLength())
+	                .header("Content-Type", contentType)
+	                .body(resource);
+    	}catch (Exception e) {
```

Is this `catch (Exception e)` here because it's called from the catch block of `getAvatar()`? Won't it slurp the exception that would be thrown on other cases?

---

Yes, and remember that's when the controller is requesting an avatar, so we want to ignore exceptions in this case.